src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHost.cs

```diff
@@ -158,12 +158,19 @@ internal async Task RunAsyncCore(CancellationToken cancellationToken, WebAssembl
                 var loggerFactory = Services.GetRequiredService<ILoggerFactory>();
                 _renderer = new WebAssemblyRenderer(Services, loggerFactory);
 
-                var rootComponents = _rootComponents;
-                for (var i = 0; i < rootComponents.Length; i++)
+                WebAssemblyCallQueue.Schedule((_rootComponents, _renderer), static state =>
                 {
-                    var rootComponent = rootComponents[i];
-                    await _renderer.AddComponentAsync(rootComponent.ComponentType, rootComponent.Selector, rootComponent.Parameters);
-                }
+                    var (rootComponents, renderer) = state;
+                    for (var i = 0; i < rootComponents.Length; i++)
+                    {
+                        var rootComponent = rootComponents[i];
+
+                        // No need to await because we want the root components to be added in parallel (e.g.,
+                        // not waiting for each other's OnInitializedAsync tasks), and any async exceptions will
+                        // be handled by normal component lifecycle mechanisms.
+                        _ = renderer.AddComponentAsync(rootComponent.ComponentType, rootComponent.Selector, rootComponent.Parameters);
```

Comment:
This changes the semantics for line 175 below. 

We should also look at what we do on Blazor server and Desktop and see if we do it in sequence or in parallel. I think we should be consistent across platforms